src/v/kafka/utils/BUILD

```diff
@@ -0,0 +1,23 @@
+load("//bazel:build.bzl", "redpanda_cc_library")
+
+redpanda_cc_library(
+    name = "utils",
```

Comment:
Can we name this specifically. Naming this the same as the package will encourage adding on to this target, where util targets should be broken up (I haven't yet found time to break up the transforms package).

Anyways let's rename to `txn_reader`

Reply:
I was debating whether to do it or not and ended up not doing it.. guess it makes sense in the spirit of breaking up targets for bazel.